src/css/node.rs

```diff
@@ -0,0 +1,405 @@
+        Block { children }
+    }
+
+    pub fn add_child(&mut self, child: BlockChild) {
+        self.children.push(child)
+    }
+}
+
+#[derive(PartialEq, Default)]
+pub struct Identifier {
+    name: String,
+}
+
+impl Debug for Identifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.name)
+    }
+}
+
+impl Identifier {
+    pub fn new(name: String) -> Identifier {
+        Identifier { name }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CDC;
+#[derive(Debug, PartialEq)]
+pub struct CDO;
+
+#[derive(Debug, PartialEq)]
+pub struct Combinator {
+    name: String,
+}
+
+#[derive(Debug, PartialEq, Default)]
+pub struct Declaration {
+    pub important: bool,
+    pub property: String,
+    pub value: ValueList,
+}
+
+impl Declaration {
+    pub fn new(property: String, value: ValueList) -> Declaration {
+        Declaration {
+            important: false,
+            property,
+            value,
+        }
+    }
+
+    pub fn set_important_as(&mut self, important: bool) {
+        self.important = important;
+    }
+
+    pub fn set_property(&mut self, property: String) {
+        self.property = property;
+    }
+
+    pub fn set_value(&mut self, value: ValueList) {
+        self.value = value;
+    }
+}
+
+#[derive(Debug, PartialEq, Default)]
+pub struct DeclarationList {
+    children: Vec<Declaration>,
+}
+
+impl DeclarationList {
+    pub fn new(children: Vec<Declaration>) -> DeclarationList {
+        DeclarationList { children }
+    }
+
+    pub fn add_child(&mut self, child: Declaration) {
+        self.children.push(child)
+    }
+}
+
+#[derive(PartialEq, Default)]
+pub struct Dimension {
+    value: String,
+    unit: Option<String>,
+}
+
+impl Debug for Dimension {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}{}", self.value, self.unit.clone().unwrap_or_default())
+    }
+}
+
+impl Dimension {
+    pub fn new(value: String, unit: Option<String>) -> Dimension {
+        Dimension { value, unit }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum MediaFeatureValue {
+    Identifier(Identifier),
+    Number(CssNumber),
+    Dimension(Dimension),
+    Ratio(Ratio),
+    Function(Function),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct MediaFeature {
+    name: String,
+    value: Option<MediaFeatureValue>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum FunctionChild {
+    Identifier(Identifier),
+    Operator(Operator),
+    Percentage(Percentage),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Function {
+    name: String,
+    children: Vec<FunctionChild>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Hash {
+    value: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Layer {
+    name: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct LayerList {
+    children: Vec<Layer>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum MediaQueryChild {
+    Identifier(Identifier),
+    MediaFeature(MediaFeature),
+}
```

Comment:
@Kiyoshika this is the pattern I was thinking of, where you have an enum variant and then an inner struct with the same name.  We're almost there with [NodeData](https://github.com/gosub-browser/gosub-engine/blob/9fa2622869f5723d151ecb6e05aa454992008d76/src/html5_parser/node.rs#L30-L35), but not quite.

Reply:
@emwalker I almost followed this but thought it might be confusing having two structs named `Document` in the code base that served different purposes, which is why I settled on `...Data` suffix. Although my opinion is not that strong so I wouldn't mind refactoring that at a later point

Reply:
(not to mention the `DocumentData` actually doesn't really do anything anyways)

Reply:
Rust's module system allows the reuse of struct names in different contexts as many times as you need.  If you ever have to use them in the same file, you can do:
```rust
use crate::{
  dom::Element,
  node::data::element::Element as ElementData,
};
```

So there's less of a need to coordinate struct names globally like you would in some other languages.  (Now that I'm looking at this, I'm wondering why `node::` isn't under `dom::`.)

Reply:
the reason is that the current "document" and nodes are an intermediate step before generating the DOM.   It made more sense in getting things up and running quickly.. but will eventually (and sooner rather than later), be moved to generating DOM as soon as possible, most likely removing the whole intermediate step..